File: InsertionSort/src/SelectionSort.java

public class SelectionSort {
	
	public static void main(String [] args)
	{
		System.out.println("Lets go for Selection sort now");
		int tmp;
		int minIndex = 0;// find the min number and pass its index to swap the numbers
		int [] numbers = {2,4,5,1,3,8,9,2,2,10,34};
		printArray(numbers);
		//tmp = numbers[minIndex];
		for (int i = 0; i<=numbers.length-2 ;i++)
		{
			minIndex = i;
			for(int j = i; j<=numbers.length-1;j++)
			{
				
				if(numbers[minIndex] > numbers[j])
				{
					minIndex = j;
					//System.out.println("minIndex=" + minIndex);
				}
			}
				
			tmp = numbers[minIndex];
			numbers[minIndex] = numbers[i];
			numbers[i] = tmp;
		}
		System.out.println(" ");
		printArray(numbers);
		
	}
	
	
	
	public static void printArray(int[] a)
	{
		for(int s=0; s<a.length;s++)
			System.out.print(a[s] + " ");
	}

}

File: pythonscripts/selectionsort.py
def main():
    list = [32,70,-1,0,66,14,30,31,90]
    print("numbers=",selectionSort(list))

def selectionSort(numbers):
#    minIndex = 0
#    min = numbers[0]
    for j in range(0,len(numbers)-1):
        minIndex = j
#        min = numbers[minIndex]
        for i in range(j+1,len(numbers)):
#            min = numbers[j+1]
            if numbers[minIndex] > numbers[i]:
                minIndex = i
#                min = numbers[minIndex]
        tmp = numbers[minIndex]
        numbers[minIndex] = numbers[j]
        numbers[j]=tmp
    return numbers



if __name__=="__main__": main()
